river/Config.zig

```diff
@@ -94,3 +94,16 @@ pub fn deinit(self: Self) void {
     self.float_filter.deinit();
     self.csd_filter.deinit();
 }
+
+pub var focused_view_opacity_delta : f32 = 0.0;
+var _last_get_time : u64 = 0;
+pub var focused_view_opacity : f32 = 1.0;
+pub fn get_focused_view_opacity() f32 {
+    var t : u64 = std.time.milliTimestamp();
```

Comment:
Really don't like this.

This function depends on river rendering multiple times a second. That will change in the future when river switches to damage based rendering.

The proper way would be to attach a timer to the event loop.

Reply:
Timer feels way to heavy. I think a better solution would be to provide a is_opacity_changing() function. Then in the main loop we could damage the focused window if the opacity is changing. Then everything would still work and we only render multiple times a second when we actually are changing the opacity no?

Reply:
I do not think that is a good solution. Think of screens with a high refresh rate. A timer ensures that we change the opacity a specific amount of time.

Reply:
>  Timer feels way to heavy. I think a better solution would be to provide a is_opacity_changing() function. Then in the main loop we could damage the focused window if the opacity is changing. Then everything would still work and we only render multiple times a second when we actually are changing the opacity no?

As leon says, the proper way to do this is to use a timer. There is no "main loop," code is only run in reaction to events.

Reply:
Okay, I will move it too a timer. However, how do I make a timer that runs at the refresh rate of the current focused output?

Reply:
You don't. Animations like fading usually work by having to values: `delta_x` and `delta_t`. The first is the difference between the "steps", the second is the time between them. `delta_t` should not be equal to the refresh rate. That would be total overkill, way beyond human perception. I doubt more than six steps per second are needed.

Reply:
(You will want to look at the wl_event_loop documentation. AFAIK it has a nice function to insert a generic timer.)